fastapi_keycloak/api.py

```diff
@@ -423,6 +423,198 @@ def delete_role(self, role_name: str) -> dict:
+    def get_all_groups(self) -> List[KeycloakGroup]:
+        """ Get all base groups of the Keycloak realm
+
+        Returns:
+            List[KeycloakGroup]: All base groups of the realm
+
+        Raises:
+            KeycloakError: If the resulting response is not a successful HTTP-Code (>299)
+        """
+        return self._admin_request(url=self.groups_uri, method=HTTPMethod.GET)     
+        
+    @result_or_error(response_model=KeycloakGroup, is_list=True)
+    def get_groups(self, group_names: List[str]) -> List[KeycloakGroup] or None:
+        """ Returns full entries of base Groups based on group names
+
+        Args:
+            group_names List[str]: Groups that should be looked up (names)
+
+        Returns:
+             List[KeycloakGroup]: Full entries stored at Keycloak. Or None if the list of requested groups is None
+
+        Raises:
+            KeycloakError: If the resulting response is not a successful HTTP-Code (>299)
+        """
+        if group_names is None:
+            return None
+        groups = self.get_all_groups()
+        return list(filter(lambda group: group.name in group_names, groups))   
+    
+    def get_subgroups(self, group: KeycloakGroup, path: str):
+        """Utility function to iterate through nested group structures
+        
+        Args: 
+            group KeycloakGroup: Group Representation
+            path str: Subgroup path
+            
+        Returns:
+            KeycloakGroup: Keycloak group representation or none if not exists
+        """
+        for subgroup in group.subGroups:
+            if subgroup.path == path:
+                return subgroup
+            elif subgroup.subGroups:
+                for subgroup in group.subGroups:
+                    result = self.get_subgroups(subgroup, path)
+                    if result:
+                        return result
+        # Went through the tree without hits
+        return None  
+    
+    @result_or_error(response_model=KeycloakGroup)
+    def get_group_by_path(self, path: str, search_in_subgroups=True) -> KeycloakGroup or None:
+        """ Return Group based on path
+
+        Args:
+            path str: Path that should be looked up
+            search_in_subgroups bool: Whether to search in subgroups
+
+        Returns:
+            KeycloakGroup: Full entries stored at Keycloak. Or None if the path not found       
+
+        Raises:
+            KeycloakError: If the resulting response is not a successful HTTP-Code (>299)
+        """
+        groups = self.get_all_groups()
+        
+        for group in groups:
+            if group.path == path:
+                return group
+            elif search_in_subgroups and group.subGroups:
+                for group in group.subGroups:
+                    if group.path == path:
+                        return group
+                    res = self.get_subgroups(group, path)
+                    if res != None:
+                        return res        
```

Comment:
Let's stick to the term above (whatever we decide on)

Reply:
maybe also say 
```python
if res is not None:
```
since this is more pythonic?

Reply:
> maybe also say
> 
> ```python
> if res is not None:
> ```
> 
> since this is more pythonic? Could also apply for the term above to make it more explicit

It really gets better this way